sdk/ml/azure-ai-ml/azure/ai/ml/_internal/_schema/component.py

```diff
@@ -36,9 +38,7 @@ class NodeType:
     HEMERA = "HemeraComponent"
     AE365EXEPOOL = "AE365ExePoolComponent"
     IPP = "IntellectualPropertyProtectedComponent"
-    SPARK = "spark"
-    # hack: used as key in factory create function dict to distinguish between spark and internal spark
-    INTERNAL_SPARK = "internal_spark"
+    INTERNAL_SPARK = "InternalSpark"
```

Comment:
Maybe call this SparkComponent like other internal types?

Reply:
its value need to be different from 3p spark, so it will be different from what we have filled in the exact column. Let me switch the variable name back. And maybe value it as DummySpark